add guide to build docker image of cve_bin_tool (#1141)
intel/cve-bin-tool: doc/how_to_guides/create_docker_file.md

├── MANIFEST.in
├── README.md
├── requirements.txt
├── setup.py
├── spdx_header.txt
└── test
```
Start by setting the python base image for our container, which for this example will be the `python:3.8-slim-buster`.  
You can choose whatever suits you the best. Some python base images can be found [here](https://hub.docker.com/_/python)  
```
FROM python:3.8-slim-buster
```

Add some python environment variable to get rid of .pyc generation and to turn off buffering. (Optional)
```
# Keeps Python from generating .pyc files in the container
ENV PYTHONDONTWRITEBYTECODE=1

# Turns off buffering for easier container logging
ENV PYTHONUNBUFFERED=1
```

Now, set the working directory for our project inside our docker container. Let's name it 'cve_bin_tool'.
```
WORKDIR /cve_bin_tool
```

Install dependencies - The packages you need to run CVE Binary Tool are listed in the requirements.txt file in the main cve-bin-tool directory. You can install all of them using the following pip command.
```
RUN python -m pip install -r requirements.txt
```
Copy the project inside our docker container. This will copy all the contents of cve_bin_tool inside our docker image.
```
COPY . .
```

> Let's create a non-root user for our image and give it the neccessary permissions to work on.

Reviewer: ```suggestion
Let's create a non-root user for our image and give it the necessary permissions to work with.
```

```
# Creates a non-root user and adds permission to access the /cve_bin_tool folder
RUN useradd cvebintooluser && chown -R cvebintooluser /cve_bin_tool
USER cvebintooluser 
```

> Finally, lets run CVE Binary Tool to fetch CVE Data from NVD.

Reviewer: ```suggestion
Finally, let's run CVE Binary Tool to fetch CVE data from NVD.
```

```
RUN python -m cve_bin_tool.cli 
```

> Alternatively, you can setup your docker container to fetch the NVD Data every time you open your docker image by using:

Reviewer: ```suggestion
Alternatively, you can setup your Docker container to fetch the NVD data every time you open your Docker image by using:
```

```
CMD ["python","-m","cve_bin_tool.cli","scan_path"]
```

And if you want to do everything manually, you can always use this to access your docker image.
```
CMD ["/bin/bash"]
```

Your Dockerfile will look something like this:
```
FROM python:3.8-slim-buster

# Keeps Python from generating .pyc files in the container
ENV PYTHONDONTWRITEBYTECODE=1

# Turns off buffering for easier container logging
ENV PYTHONUNBUFFERED=1

WORKDIR /cve_bin_tool

> # Install pip requirements

Reviewer: ```suggestion
# Install cve-bin-tool requirements
```

COPY requirements.txt requirements.txt
RUN python -m pip install -r requirements.txt

> # Copy cve_bin_tool directory to project

Reviewer: ```suggestion
# Copies cve_bin_tool directory to container
```

COPY . .

# Creates a non-root user and adds permission to access the /cve_bin_tool folder
RUN useradd cvebintooluser && chown -R cvebintooluser /cve_bin_tool
USER cvebintooluser

RUN python -m cve_bin_tool.cli 

CMD ["/bin/bash"]

```
# How to build the docker image

> For building the image simply run 

Reviewer: ```suggestion
To build the image simply run 
```

```
docker build ${cve-bin-tool} -t "cve_bin_tool"
```

> Here ${cve-bin-tool} is the directory path to cve_bin_tool and "cve_bin_tool" is the name of your docker image.

Reviewer: ```suggestion
Here `${cve-bin-tool}` is the directory path to cve_bin_tool and "cve_bin_tool" is the name of your Docker image.
```

# How to use the docker image
If you want to mount a directory to your docker image while initializing it, simply use:
```
docker run -it -v ${directory_to_scan}:${directory_in_docker} cve_bin_tool
```

> Here `directory_to_scan` is the path to the directory in your host machine which you want to scan.

Reviewer: ```suggestion
Here `${directory_to_scan}` is the path to the directory on your host machine which you want to scan.
```
Just to make it the same as `${cve-bin-tool}` above.

> And `directory_in_docker` is the path in the docker image where you mount the above directory.

Reviewer: ```suggestion
And `${directory_in_docker}` is the path in the Docker image where you mount the above directory.
```
